pkg/charts/defaults.go

```diff
@@ -0,0 +1,265 @@
+	}
+
+	proxy, err := defaults.Table("Proxy")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyImage, err := defaults.Table("Proxy.Image")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyInitImage, err := defaults.Table("ProxyInit.Image")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyInitResourcesCPU, err := defaults.Table("ProxyInit.Resources.CPU")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyInitResourcesMemory, err := defaults.Table("ProxyInit.Resources.Memory")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyPorts, err := defaults.Table("Proxy.Ports")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyResourcesCPU, err := defaults.Table("Proxy.Resources.CPU")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyResourcesMemory, err := defaults.Table("Proxy.Resources.Memory")
+	if err != nil {
+		return nil, err
+	}
+
+	defaultValues := &DefaultValues{
+		ControllerReplicas:               uint(defaults["ControllerReplicas"].(float64)),
+		ControllerLogLevel:               defaults["ControllerLogLevel"].(string),
+		ControllerUID:                    int64(defaults["ControllerUID"].(float64)),
+		EnableExternalProfiles:           proxy["EnableExternalProfiles"].(bool),
+		EnableH2Upgrade:                  defaults["EnableH2Upgrade"].(bool),
+		ImagePullPolicy:                  defaults["ImagePullPolicy"].(string),
+		IdentityTrustDomain:              identity["TrustDomain"].(string),
+		IdentityIssuerClockSkewAllowance: identityClockSkewAllowance,
+		IdentityIssuerIssuanceLifetime:   identityIssuanceLifetime,
+		OmitWebhookSideEffects:           defaults["OmitWebhookSideEffects"].(bool),
+		PrometheusImage:                  defaults["PrometheusImage"].(string),
+		ProxyAdminPort:                   uint(proxyPorts["Admin"].(float64)),
+		ProxyControlPort:                 uint(proxyPorts["Control"].(float64)),
+		ProxyCPULimit:                    proxyResourcesCPU["Limit"].(string),
+		ProxyCPURequest:                  proxyResourcesCPU["Request"].(string),
+		ProxyImageName:                   proxyImage["Name"].(string),
+		ProxyInboundPort:                 uint(proxyPorts["Inbound"].(float64)),
+		ProxyInitImageName:               proxyInitImage["Name"].(string),
+		ProxyInitCPULimit:                proxyInitResourcesCPU["Limit"].(string),
+		ProxyInitCPURequest:              proxyInitResourcesCPU["Request"].(string),
+		ProxyInitMemoryLimit:             proxyInitResourcesMemory["Limit"].(string),
+		ProxyInitMemoryRequest:           proxyInitResourcesMemory["Request"].(string),
+		ProxyLogLevel:                    proxy["LogLevel"].(string),
+		ProxyMemoryLimit:                 proxyResourcesMemory["Limit"].(string),
+		ProxyMemoryRequest:               proxyResourcesMemory["Request"].(string),
+		ProxyOutboundPort:                uint(proxyPorts["Outbound"].(float64)),
+		ProxyUID:                         int64(proxy["UID"].(float64)),
+		WebhookFailurePolicy:             defaults["WebhookFailurePolicy"].(string),
+	}
+
+	if ha {
```

Comment:
in the non-ha case above, i liked how your grouped all the `defaults.Table()` calls together and the the assignment into `defaultValues` at the end. would you mind making this consistent with that?